addons/account/models/res_config_settings.py

```diff
@@ -57,10 +57,21 @@ class ResConfigSettings(models.TransientModel):
     tax_exigibility = fields.Boolean(string='Cash Basis', related='company_id.tax_exigibility')
     tax_cash_basis_journal_id = fields.Many2one('account.journal', related='company_id.tax_cash_basis_journal_id', string="Tax Cash Basis Journal")
     account_hide_setup_bar = fields.Boolean(string='Hide Setup Bar', related='company_id.account_setup_bar_closed',help="Tick if you wish to hide the setup bar on the dashboard")
+    vendor_bill_alias_prefix = fields.Char('Default Alias Name for Vendor Bills')
+    account_use_mailgateway = fields.Boolean(string='Email your Vendor Bills', config_parameter='account.account_use_mailgateway', default=True)
+
+    @api.model
+    def get_values(self):
```

Comment:
can't that be handled by config_parameter='account.mail_alias_vendor_bills' on the field?? what's the difference and why don't you do the same for account_use_mailgateway field?

Reply:
Actually, I think these options need to be company dependent, since you'll want to give a dedicated alias for each company in case of multi-company database

So I think you need to replace all this with regular fields stored on the company

Reply:
Ok, I'll make it company dependant considering multi-company scenario

Reply:
ok for me

Reply:
@qdp-odoo in reply to your comment,
"can't that be handled by config_parameter='account.mail_alias_vendor_bills' on the field?? what's the difference and why don't you do the same for account_use_mailgateway field?"

we tried to implement it as we have implemented the same thing in <a href="https://github.com/odoo/odoo/blame/master/addons/hr_expense/models/res_config_settings.py">hr_expense</a> and <a href="https://github.com/odoo/odoo/blob/master/addons/crm/models/res_config_settings.py#L10">crm</a> modules, but yes, it can be handled if we add `config_parameter='account.mail_alias_vendor_bills'` on field `vendor_bill_alias_prefix`. So do you want me to handle this with `config_parameter` on that field ?

Reply:
I guess, yes... since it would make less code (no need to overwrite get_values() and set_values() for that field). Right?

Reply:
get_values() code won't be needed that way(with cost of one extra ir_config.parameter record) but we need to update `mail_alias_vendor_bills` <a href="https://github.com/odoo/odoo/pull/22158/files#diff-a65bf9be364df91b6656a8b19d6af619R6">record</a>'s value from set_values() anyway.

Reply:
@qdp-odoo , no need to store mail alias in confirm setting.
alias_prefix is stored in 'mail.alias' model. so we need set_value/get_value to sync alias_prefix with 'mail.alias' model

see https://github.com/odoo/odoo/blob/master/addons/crm/models/res_config_settings.py#L54: set_values method in crm setting model. here alias record is created in mail.alias model 

and get_values, to get alias from mail_alias model ( https://github.com/odoo/odoo/blob/master/addons/crm/models/res_config_settings.py#L19)

so need set_values and get_values

cc @tde-banana-odoo